.github/workflows/push.yml

```diff
@@ -38,105 +38,103 @@ jobs:
-          SHORT_BRANCH_NAME=${SHORT_BRANCH_NAME//\//-}
-          SHORT_BRANCH_NAME=${SHORT_BRANCH_NAME//./-}
-          SHORT_BRANCH_NAME=${SHORT_BRANCH_NAME//!/-}
-          echo "GHA_DOCKER_PUSH_BRANCH_TAG=$SHORT_BRANCH_NAME" >> $GITHUB_ENV
-      - id: set-date-env
+          BRANCH_NAME=${BRANCH_NAME:0:43} # truncate to max_len - len(.YYYYMMDD-SHA1234567)
+          BRANCH_NAME=$(echo "$BRANCH_NAME" | sed s'/[-]$//') # remove trailing -
+          BRANCH_NAME=$(echo "$BRANCH_NAME" | tr '[:upper:]' '[:lower:]' | tr -d '[ÆØÅæøå]') # to ASCII lower case
+          BRANCH_NAME=${BRANCH_NAME//\//-} # replace / with -
+          BRANCH_NAME=${BRANCH_NAME//./-}  # replace . with -
+          BRANCH_NAME=${BRANCH_NAME//!/-}  # replace ! with -
+          echo "GHA_DOCKER_PUSH_BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_ENV
+      - id: verify-cloud-provider
         shell: bash
-        run: echo "GHA_DOCKER_PUSH_DATE=$(date +'%Y.%m.%d')" >> $GITHUB_ENV
-      - id: set-docker-tag-from-input-env
+        if: env.GHA_DOCKER_PUSH_CLOUD_PROVIDER != 'GCP' && env.GHA_DOCKER_PUSH_CLOUD_PROVIDER != 'Azure'
+        run: |
+          echo "cloud_provider can only be GCP or Azure"
+          exit 1
+      - id: set-image-artifact-name
         shell: bash
+        if: env.GHA_DOCKER_PUSH_IMAGE_NAME == 'repo_name'
         run: |
-          if [[ "${{ env.GHA_DOCKER_PUSH_DOCKER_TAG }}" = "docker_tag" ]]; then
-            SHA=${{ github.sha }}
-            SHORT_SHA=${SHA:0:7}
-            echo "GHA_DOCKER_PUSH_DOCKER_TAG=${{ env.GHA_DOCKER_PUSH_BRANCH_TAG }}.${{ env.GHA_DOCKER_PUSH_DATE }}-SHA$SHORT_SHA" >> $GITHUB_ENV
-          fi
-      - id: set-repository-env
+          echo "GHA_DOCKER_PUSH_IMAGE_NAME=${GITHUB_REPOSITORY#*/}" >> $GITHUB_ENV
+      - id: set-image-tag
         shell: bash
+        if: env.GHA_DOCKER_PUSH_IMAGE_TAG  == 'image_tag'
         run: |
-          if [[ "${{ inputs.repository_type }}" = "GCR" ]]; then
-            echo "GHA_DOCKER_PUSH_META_IMAGES=eu.gcr.io/${{ env.GHA_DOCKER_PUSH_GCR_PROJECT_ID }}/${{ env.GHA_DOCKER_PUSH_IMAGE_ARTIFACT_NAME }}" >> $GITHUB_ENV
-          elif [[ "${{ inputs.repository_type }}" = "ACR" ]]; then
-            echo "GHA_DOCKER_PUSH_META_IMAGES=acrentur001.azurecr.io/${{ env.GHA_DOCKER_PUSH_IMAGE_ARTIFACT_NAME }}" >> $GITHUB_ENV
-          fi
+          SHA=${{ github.sha }}
+          SHORT_SHA=${SHA:0:7}
+          echo "GHA_DOCKER_PUSH_IMAGE_TAG=${{ env.GHA_DOCKER_PUSH_BRANCH_NAME }}.${{ env.GHA_DOCKER_PUSH_DATE }}-SHA$SHORT_SHA" >> $GITHUB_ENV
+      - id: set-registry-azure
+        shell: bash
+        if: env.GHA_DOCKER_PUSH_CLOUD_PROVIDER  == 'Azure'
+        run: |
+          echo "GHA_DOCKER_PUSH_DOCKER_REGISTRY=acrentur001.azurecr.io" >> $GITHUB_ENV
+      - id: set-registry-gcp
+        shell: bash
+        if: env.GHA_DOCKER_PUSH_CLOUD_PROVIDER  == 'GCP'
+        run: |
+          echo "GHA_DOCKER_PUSH_DOCKER_REGISTRY=eu.gcr.io/entur-system-1287" >> $GITHUB_ENV
```

Comment:
registry, org secrets? (todo?)

Reply:
What do you think @Spikra @bvestli? I agree.
Then we could simply redirect this to whatever secret name we decide for ACR and GCR (and cleanly swap when we go GAR).

Reply:
Agree on that 👍